ui/hooks/useAlerts.test.ts

```diff
@@ -156,16 +163,42 @@ describe('useAlerts', () => {
           },
         },
       });
-      expect(result1.current.hasUnconfirmedFieldDangerAlerts).toEqual(false);
+      expect(result.current.hasUnconfirmedFieldDangerAlerts).toEqual(false);
     });
   });
 
   describe('generalAlerts', () => {
-    it('returns general alerts', () => {
-      const expectedGeneralAlerts = alertsMock.find(
-        (alert) => alert.key === dataAlertKeyMock,
+    it('returns general alerts sorted by severity', () => {
+      const warningGeneralAlert = {
+        key: dataAlertKeyMock,
+        severity: Severity.Warning as AlertSeverity,
+        message: 'Alert 2',
+      };
+      const expectedGeneralAlerts = [
+        warningGeneralAlert,
+        {
+          ...warningGeneralAlert,
+          severity: Severity.Info as AlertSeverity,
+          message: 'Alert 3',
+        },
+        {
+          ...warningGeneralAlert,
+          severity: Severity.Danger as AlertSeverity,
+          message: 'Alert 1',
+        },
+      ];
+
+      const result = renderAndReturnResult(undefined, {
+        confirmAlerts: {
+          alerts: {
+            [ownerIdMock]: expectedGeneralAlerts,
+          },
+          confirmed: {},
+        },
+      });
+      expect(result.current.generalAlerts).toEqual(
+        sortAlertsBySeverity(expectedGeneralAlerts),
```

Comment:
If this is a requirement of the logic, should we instead test the order directly, rather than exporting code that is private to the hook?

Reply:
Right, set the `sortAlertsBySeverity` to private and tested the logic directly [here](https://github.com/MetaMask/metamask-extension/pull/27339/commits/351bbed65cca27291a9f9515f1ca812813835db7).